### PR TITLE
#33 Feature: User Update Profile

### LINKED ISSUE #32: Hide coverage, Our repo is covered with HTML

### PR DESCRIPTION
**Pull request by kagaba-etienne**
#### What does this PR do?

- Creates a UI to enable user in profile updating
- Adds necessary intergration with backend to facilitate user in editing his/her profile
- Closes #32 

#### Description of Task to be completed?

> Users should be able to amend/edit their profiles depending on a change of preference to the entire disclosure of the individual user
>
> As a User
>
> I want to get my profile created once I register for the application.
I want my profile to display my Personal Information name, email,gender, birthdate, preferred language, preferred currency ,where you live, Billing Address (check the right fields of the billing address)
>
> Acceptance Criteria
>
> I want to be able to Edit my profile, but I should NOTbe able to edit another users profile
> Scenario:
> Given User registers to be a member of the App.
> Registration is done.
> User Profile is created.
> A timestamp is associated to the profile

#### How should this be manually tested?
```
git clone https://github.com/atlp-rwanda/ecommerce-app-predators-fe.git
cd ecommerce-app-predators-fe
git checkout ft-user-update-profile
npm i
npm run dev
```

#### What are the relevant pivotal tracker/Trello stories?
[Card #33](https://trello.com/c/TmVCsXQw)

#### Screenshot
Screenshot 2023-06-19 at 18 36 45

### CODE REVIEW

**Review by Muhet — APPROVED** [review]
Great job

**Review by yobuir — APPROVED** [review]

**Review by vanessa-abewe — APPROVED** [review]

**Review by nili-alu — APPROVED** [review]
good job

File: src/redux/action/profileAction.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        return response.data;
+      } catch (error: any) {
+        return thunkAPI.rejectWithValue(error.response.data);
+      }
+    }
+  )
+
+export const updateProfile = createAsyncThunk(
+    "profile/update",
+    async (profile: any, thunkAPI) => {
```
Comment (chanceDira) [inline src/redux/action/profileAction.tsx]: can you use interface here

**Review by chanceDira — CHANGES_REQUESTED** [review]
Nice job @kagaba-etienne just one thing

File: src/redux/action/profileAction.tsx (hunk 1 above)
Comment (kagaba-etienne) [reply src/redux/action/profileAction.tsx]: okay sure!

**Review by faid-terence — APPROVED** [review]

**Review by chanceDira — APPROVED** [review]
Good

**Pull request merged**